File: python23/palindrome.py
#Problem 6

def palindrome(one):

    check_palindrome = True

    for i in range (len(one)//2):
        if one[i] != one[len(one)-i-1]:
           check_palindrome = False

    return check_palindrome


example_list = ['r', 'b', 'c', 'b', 'r']

value = palindrome(example_list)

if value == True:
    print("Palindrome")

else:
    print("Not Palindrome")

File: java23/Palindrome.java
//Problem 6

package java23;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Palindrome {

    public static boolean isPalindrome(List<String> list) {

        for (int i = 0; i < list.size() / 2; i++) {
            if (!list.get(i).equals(list.get(list.size() - i - 1))) {
                return false;
            }
        }
        return true;

    }

    public static void main(String[] args) {

        List<String> list = new ArrayList<>(Arrays.asList("a", "b", "c", "b", "a"));
        boolean check_palindrome = isPalindrome(list);

        if (check_palindrome == true) {
            System.out.println("Palindrome");
        } else {
            System.out.println("Not Palindrome");
        }
    }
}